tests/integration_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,182 @@
+import bq_utils
+from utils.bq import get_client
+import gcs_utils
+from tests import test_util
+from retraction import retract_deactivated_pids
+from io import open
+from constants.cdr_cleaner import clean_cdr as clean_consts
+
+TABLE_ROWS_QUERY = 'SELECT * FROM {dataset_id}.__TABLES__ '
+
+EXPECTED_ROWS_QUERY = ('SELECT * '
+                       'FROM {dataset_id}.{table_id} '
+                       'WHERE person_id = {pid} '
+                       'AND {date_column} >= (SELECT deactivated_date '
+                       'FROM {dataset_id}.{pid_table_id} '
+                       'WHERE person_id = {pid})')
+
+INSERT_PID_TABLE = ('INSERT INTO {dataset_id}.{pid_table_id} '
+                    '(person_id, deactivated_date) '
+                    'VALUES{person_research_ids}')
+
+
+class RetractDataBqTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.hpo_id = 'fake'
+        self.project_id = app_identity.get_application_id()
+        self.fake_project_id = 'fake-project-id'
+        self.bq_dataset_id = bq_utils.get_dataset_id()
+        self.ticket_number = 'DCXXX'
+        self.pid_table_id = 'pid_table'
+        self.deactivated_ehr_participants = [(1, '2007-01-01'),
+                                             (2, '2008-02-01'),
+                                             (1234567, '2019-01-01')]
+
+    @mock.patch(
+        'retraction.retract_deactivated_pids.get_date_info_for_pids_tables')
+    @mock.patch('bq_utils.list_datasets')
+    def test_integration_queries_to_retract_from_fake_dataset(
+        self, mock_list_datasets, mock_retraction_info):
+        mock_list_datasets.return_value = [self.bq_dataset_id]
+        d = {
+            'project_id': [
+                self.project_id, self.project_id, self.project_id,
+                self.project_id, self.project_id, self.project_id
+            ],
+            'dataset_id': [
+                self.bq_dataset_id, self.bq_dataset_id, self.bq_dataset_id,
+                self.bq_dataset_id, self.bq_dataset_id, self.bq_dataset_id
+            ],
+            'table': [
+                'fake_condition_occurrence', 'fake_drug_exposure',
+                'fake_measurement', 'fake_observation',
+                'fake_procedure_occurrence', 'fake_visit_occurrence'
+            ],
+            'date_column': [
+                'condition_end_date', 'drug_exposure_end_date',
+                'measurement_date', 'observation_date', 'procedure_date',
+                'visit_end_date'
+            ]
+        }
+        retraction_info = pd.DataFrame(data=d)
+        mock_retraction_info.return_value = retraction_info
+
+        # Create and load person_ids and deactivated_date to pid table
+        bq_utils.create_table(self.pid_table_id,
```

Comment:
Since this is new work, this should be using the bigquery client library instead of `bq_utils`.

Reply:
when trying to convert this to use the bq utility `bq.create_tables`, I found that this function uses the `fields_for(table_name)` in `resources.py`. Am I ok to add a JSON file to the fields folder for this test table? Or should `bq.create_tables` be modified somehow to not rely on those schemas?

Reply:
I absolutely would not add a test table schema to our fields paths.  You are right to consider altering `bq.create_tables`.  You could add an optional `fields` parameter to the signature and pass that value to the `get_table_schema` function.  Again, the `fields` parameter should be optional.  If it exists, it should be used and if it doesn't it should use the `fields_for` function.  Does this make sense?

Reply:
Yes, thank you! I've updated `utils.bq` and tested locally through my integration test. Let me know if any changes need to be made.